wgengine/magicsock/magicsock.go

```diff
@@ -1236,6 +1251,15 @@ func (c *Conn) mkReceiveFunc(ruc *RebindingUDPConn, healthItem *health.ReceiveFu
 				if neterror.PacketWasTruncated(err) {
 					continue
 				}
+				if neterror.SocketWasReset(err) {
+					// Wrap the error in a permanentNetError so that Wireguard
+					// doesn't keep trying to read packets from us.
+					connErr = permanentNetError{error: err}
+					c.logf("magicsock: receive: rebind required due to socket reset: %v", err)
+					c.Rebind()
+					return 0, connErr
```

Comment:
Looking into this more, I don't think a permanent error is the right thing when combined with rebinding.  The magicsock rebinds under the covers as far as the Wireguard device is concerned. If we return a permanent error, Wireguard will stop reading from this function, and has to be stopped and restarted. I guess just return the original error so it gets logged.